storage/remote/queue_manager.go

```diff
@@ -1921,11 +1924,35 @@ func populateV2TimeSeries(symbolTable *writev2.SymbolsTable, batch []timeSeries,
 			pendingData[nPending].Metadata.UnitRef = symbolTable.Symbolize(d.metadata.Unit)
 			nPendingMetadata++
 		} else {
+			metricType := writev2.Metadata_METRIC_TYPE_UNSPECIFIED
+			var unit string
+			if enableTypeAndUnitLabels {
+				typeValue := d.seriesLabels.Get("__type__")
+				unit = d.seriesLabels.Get("__unit__")
+				// convert string type to MetricMetadata_MetricType
+				switch typeValue {
+				case "counter":
+					metricType = writev2.Metadata_METRIC_TYPE_COUNTER
+				case "gauge":
+					metricType = writev2.Metadata_METRIC_TYPE_GAUGE
+				case "histogram":
+					metricType = writev2.Metadata_METRIC_TYPE_HISTOGRAM
+				case "summary":
+					metricType = writev2.Metadata_METRIC_TYPE_SUMMARY
+				case "info":
+					metricType = writev2.Metadata_METRIC_TYPE_INFO
+				case "stateset":
+					metricType = writev2.Metadata_METRIC_TYPE_STATESET
+				default:
+					metricType = writev2.Metadata_METRIC_TYPE_UNSPECIFIED
+				}
```

Comment:
Is that the correct way to convert the string that we are getting from labels to `MetricMetadata_MetricType`?

Reply:
Perhaps, but we have a rdy helper for it: https://github.com/prometheus/prometheus/pull/17033#discussion_r2293877864